From 0f5eee614ba0a4bb2775fd5482401c6777db97eb Mon Sep 17 00:00:00 2001
From: Marcelo Vanzin <vanzin@cloudera.com>
Date: Wed, 5 Nov 2014 11:45:46 -0800
Subject: [PATCH 001/201] CLOUDERA-BUILD. Changes for CDH build.

Adjusts dependency versions, adds Cloudera repos, and triggers
all needed profiles based on the "cdh.build" property.

(cherry picked from commit 018bcc32602c05ebb8b59d7bed8158cf25b00ca4)
---
 assembly/pom.xml |   30 ++++++++++++++++++++++
 examples/pom.xml |   25 ++++++++++++++++++
 pom.xml          |   74 ++++++++++++++++++++++++++++++++++++++----------------
 3 files changed, 107 insertions(+), 22 deletions(-)

diff --git a/assembly/pom.xml b/assembly/pom.xml
index fbabaa5..a11b8d3 100644
--- a/assembly/pom.xml
+++ b/assembly/pom.xml
@@ -168,6 +168,11 @@
   <profiles>
     <profile>
       <id>yarn</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <dependencies>
         <dependency>
           <groupId>org.apache.spark</groupId>
@@ -178,6 +183,11 @@
     </profile>
     <profile>
       <id>hive</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <dependencies>
         <dependency>
           <groupId>org.apache.spark</groupId>
@@ -188,6 +198,11 @@
     </profile>
     <profile>
       <id>hive-thriftserver</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <dependencies>
         <dependency>
           <groupId>org.apache.spark</groupId>
@@ -237,18 +252,33 @@
     <!-- Profiles that disable inclusion of certain dependencies. -->
     <profile>
       <id>hadoop-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <hadoop.deps.scope>provided</hadoop.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>hive-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <hive.deps.scope>provided</hive.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>parquet-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <parquet.deps.scope>provided</parquet.deps.scope>
       </properties>
diff --git a/examples/pom.xml b/examples/pom.xml
index 37b15bb..e8b10c0 100644
--- a/examples/pom.xml
+++ b/examples/pom.xml
@@ -401,30 +401,55 @@
     <!-- Profiles that disable inclusion of certain dependencies. -->
     <profile>
       <id>flume-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <flume.deps.scope>provided</flume.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>hadoop-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <hadoop.deps.scope>provided</hadoop.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>hbase-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <hbase.deps.scope>provided</hbase.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>hive-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <hive.deps.scope>provided</hive.deps.scope>
       </properties>
     </profile>
     <profile>
       <id>parquet-provided</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <properties>
         <parquet.deps.scope>provided</parquet.deps.scope>
       </properties>
diff --git a/pom.xml b/pom.xml
index bbee213..4c70c13 100644
--- a/pom.xml
+++ b/pom.xml
@@ -20,9 +20,9 @@
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <parent>
-    <groupId>org.apache</groupId>
-    <artifactId>apache</artifactId>
-    <version>14</version>
+    <groupId>com.cloudera.cdh</groupId>
+    <artifactId>cdh-root</artifactId>
+    <version>5.7.0-SNAPSHOT</version>
   </parent>
   <groupId>org.apache.spark</groupId>
   <artifactId>spark-parent_2.10</artifactId>
@@ -125,24 +125,24 @@
     <sbt.project.name>spark</sbt.project.name>
     <mesos.version>0.21.1</mesos.version>
     <mesos.classifier>shaded-protobuf</mesos.classifier>
-    <slf4j.version>1.7.10</slf4j.version>
+    <slf4j.version>${cdh.slf4j.version}</slf4j.version>
     <log4j.version>1.2.17</log4j.version>
-    <hadoop.version>2.2.0</hadoop.version>
-    <protobuf.version>2.5.0</protobuf.version>
+    <hadoop.version>${cdh.hadoop.version}</hadoop.version>
+    <protobuf.version>${cdh.protobuf.version}</protobuf.version>
     <yarn.version>${hadoop.version}</yarn.version>
-    <hbase.version>0.98.7-hadoop2</hbase.version>
-    <hbase.artifact>hbase</hbase.artifact>
-    <flume.version>1.6.0</flume.version>
-    <zookeeper.version>3.4.5</zookeeper.version>
-    <curator.version>2.4.0</curator.version>
-    <hive.group>org.spark-project.hive</hive.group>
+    <hbase.version>${cdh.hbase.version}</hbase.version>
+    <hbase.artifact>hbase-server</hbase.artifact>
+    <flume.version>${cdh.flume-ng.version}</flume.version>
+    <zookeeper.version>${cdh.zookeeper.version}</zookeeper.version>
+    <curator.version>${cdh.curator.version}</curator.version>
+    <hive.group>org.apache.hive</hive.group>
     <!-- Version used in Maven Hive dependency -->
-    <hive.version>1.2.1.spark</hive.version>
+    <hive.version>${cdh.hive.version}</hive.version>
     <!-- Version used for internal directory structure -->
-    <hive.version.short>1.2.1</hive.version.short>
-    <derby.version>10.10.1.1</derby.version>
-    <parquet.version>1.7.0</parquet.version>
-    <hive.parquet.version>1.6.0</hive.parquet.version>
+    <hive.version.short>1.1.0</hive.version.short>
+    <derby.version>10.11.1.1</derby.version>
+    <parquet.version>${cdh.parquet.version}</parquet.version>
+    <hive.parquet.version>${parquet.version}</hive.parquet.version>
     <jblas.version>1.2.4</jblas.version>
     <jetty.version>8.1.14.v20131031</jetty.version>
     <orbit.version>3.0.0.v201112011016</orbit.version>
@@ -150,9 +150,9 @@
     <ivy.version>2.4.0</ivy.version>
     <oro.version>2.0.8</oro.version>
     <codahale.metrics.version>3.1.2</codahale.metrics.version>
-    <avro.version>1.7.7</avro.version>
+    <avro.version>${cdh.avro.version}</avro.version>
     <avro.mapred.classifier>hadoop2</avro.mapred.classifier>
-    <jets3t.version>0.7.1</jets3t.version>
+    <jets3t.version>${cdh.jets3t.version}</jets3t.version>
     <aws.java.sdk.version>1.9.40</aws.java.sdk.version>
     <aws.kinesis.client.version>1.4.0</aws.kinesis.client.version>
     <!-- the producer is used in tests -->
@@ -168,9 +168,9 @@
     <scala.binary.version>2.10</scala.binary.version>
     <jline.version>${scala.version}</jline.version>
     <jline.groupid>org.scala-lang</jline.groupid>
-    <codehaus.jackson.version>1.9.13</codehaus.jackson.version>
-    <fasterxml.jackson.version>2.4.4</fasterxml.jackson.version>
-    <snappy.version>1.1.2</snappy.version>
+    <codehaus.jackson.version>${cdh.jackson.version}</codehaus.jackson.version>
+    <fasterxml.jackson.version>2.2.3</fasterxml.jackson.version>
+    <snappy.version>${cdh.hadoop-snappy.version}</snappy.version>
     <netlib.java.version>1.1.2</netlib.java.version>
     <calcite.version>1.2.0-incubating</calcite.version>
     <commons-codec.version>1.10</commons-codec.version>
@@ -216,6 +216,25 @@
   </properties>
   <repositories>
     <repository>
+      <id>cdh.repo</id>
+      <url>https://repository.cloudera.com/artifactory/cloudera-repos</url>
+      <name>Cloudera Repositories</name>
+      <snapshots>
+        <enabled>false</enabled>
+      </snapshots>
+    </repository>
+    <repository>
+      <id>cdh.snapshots.repo</id>
+      <url>https://repository.cloudera.com/artifactory/libs-snapshot-local</url>
+      <name>Cloudera Snapshots Repository</name>
+      <snapshots>
+        <enabled>true</enabled>
+      </snapshots>
+      <releases>
+        <enabled>false</enabled>
+      </releases>
+    </repository>
+    <repository>
       <id>central</id>
       <!-- This should be at top, it makes maven try the central repo first and then others and hence faster dep resolution -->
       <name>Maven Repository</name>
@@ -690,6 +709,7 @@
         <groupId>com.fasterxml.jackson.core</groupId>
         <artifactId>jackson-databind</artifactId>
         <version>${fasterxml.jackson.version}</version>
+        <scope>${hadoop.deps.scope}</scope>
       </dependency>
       <dependency>
         <groupId>com.fasterxml.jackson.core</groupId>
@@ -2465,6 +2485,11 @@
 
     <profile>
       <id>yarn</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <modules>
         <module>yarn</module>
         <module>network/yarn</module>
@@ -2473,6 +2498,11 @@
 
     <profile>
       <id>hive-thriftserver</id>
+      <activation>
+        <property>
+          <name>cdh.build</name>
+        </property>
+      </activation>
       <modules>
         <module>sql/hive-thriftserver</module>
       </modules>
-- 
1.7.9.5

